doc/rvv-intrinsic-spec.adoc

```diff
@@ -0,0 +1,517 @@
+
+=== Control of `eew`, `emul`
+
+The RISC-V vector intrinsics data types are strongly-typed, and hence so are the intrinsics taking these types as parameters. The vector intrinsics encode the EEW (effective-element-width) ^17^ and EMUL (effective LMUL) ^17^ in the suffix of the function name. Users can expect the results of the vector instruction intrinsics are computed under the specified EEW and EMUL.
+
+To see the full list of data types for the intrinsics, please see <<chapter-type-system>>.
+
+In the following example, the intrinsic will produce the result of `vadd.vv` with vector operands of `EEW=32` and `EMUL=1`.
+
+```
+vint32m1_t __riscv_vadd_vv_i32m1 (vint32m1_t op1, vint32m1_t op2, size_t vl);
+```
+
+In the following example, the intrinsic will produce the result of `vwadd.vv` with destinations of `EEW=32` and `EMUL=1`, and operands of `EEW=16` and `EMUL=mf2 (1/2)`.
+
+```
+vint32m1_t __riscv_vwadd_vv_i32m1 (vint16mf2_t op1, vint16mf2_t op2, size_t vl);
+```
+
+=== Control of `vl`
+
+Direct control of `vl` (vector length register) ^11^  is not exposed at the C language level. In the intrinsics that represents vector instructions, the `vl` operand indicates the active vector length valid during intrinsic execution.
+
+NOTE: Intrinsics for instructions that behave the same with different `vl` settings (e.g. `vmv.s.x`) do not have a `vl` operand.
+
+NOTE: Intrinsics will at most operate `vlmax` (derived in 3.4.2 of v-spec) elements.
+
+=== Control of `vm`
+
+Instructions that are available for masking ^7^ are also supported in the intrinsics.
+
+The intrinsics fuse the control of vector masking (`vm`) together with the control for the policy behavior in the same suffix. Please checkout <<control-to-policy>> and <<policy-naming-scheme>> for the exact suffix that specifies a masked/unmasked vector operation along with its policy behavior.
+
+
+[[control-to-policy]]
+=== Control of `vta`, `vma`
+
+The behavior of destination tail elements and destination inactive masked-off elements is controled by the `vta` and `vma` bits ^6^.
+
+Given the general assumption that target audience of the RVV intrinsics are high performance cores, and an "undisturbed" policy will generally slow down an out-of-order core, the intrinsics have a default policy scheme of tail-agnostic and mask-agnostic (that is, `vta=1` and `vma=1`).
+
+Please see <<policy-naming-scheme>> for the exact suffix that specifies the policy behavior of a vector operation.
+
+=== Control of `vxrm`
+
+For the fixed-point intrinsics, representing the instructions under v-spec section 12, the `vxrm` operand of the intrinsics indicates the rounding mode (`vxrm`) ^8^ control.
+
+The `vxrm` operand is required to be an immediate. The compiler should implement the following enum that maps to the defined rounding mode values under v-spec Table 4 ^8^.
```

Comment:
immediate -> "compile time constant" or maybe "integer constant expression" if you want to get real technical?